### ISSUE TITLE
[Feature Request]:  for ui5-input, do not autocomplete the keyword when selecting one of the suggested items

### ISSUE THREAD

**Issue by yzy415** [SAP/ui5-webcomponents#7276]
### Feature Request Description

When using ui5-input, we want to decide whether or not to autocomplete the keyword when the user selects one of the suggested items. And currently, it only fills the selected items to the input box.

In one of my team's cases, the suggested items are the results of the already entered keyword instead of the suggested keyword and we don't want the result to show in the input box. 

### Proposed Solution

keep the functionality of autocompleting as default, and support the requested feature.

### Additional Context

I'm from SF.

### Priority

High

### Privacy Policy

- [X] I’m not disclosing any internal or sensitive information.

**Comment by kgogov** [comment]
Hello @SAP/ui5-webcomponents-topic-rl, could you take a look of the following feature request?
Thank you!

Best regards,
Konstantin

**Comment by yzy415** [comment]
Hi @kgogov and @nnaydenow, will we start to implement this in September?

**Comment by yzy415** [comment]
Hi @hristop, 

It's expected that we can decide whether to autocomplete the keywords when the user clicks the suggested items. 

Best regards,
Alex Yan

**Comment by yzy415** [comment]
> Hi @hristop,
> 
> It's expected that the consumer of the Input component can decide whether to autocomplete the keywords when the user clicks the suggested items.
> 
> Best regards, Alex Yan

Could we make it by allowing the consumer to pass a boolean like 'autoComplete' as 'false' to the SuggestionItem?

**Comment by hristop** [comment]
Hi @yzy415 ,

Thanks for clarifying! 
In SAPUI5 there is the option to switch off the autocomplete for the input field, but not for specific suggestions.
We will discuss this internally on how to proceed with the request and your suggestion. 

Best Regards,
Hristo

**Comment by hristop** [comment]
Hi @yzy415 ,

The input component has noTypeahead property, which can switch on and off the type-ahead. 
Also there is the "suggestion-item-select" event, which passes the pressed item. 
Can you try combining both in order to achieve the desired functionality? 

Best Regards,
Hristo

**Comment by yzy415** [comment]
Hi @hristop ,

Thanks for your helpful advice. 

We've enabled noTypeahead and tried to suppress the default auto-complete behavior by invoking the preventDefault method on "suggestion-item-select" event. But the text of the suggestion item will still be put into the Input box when we click one of them. 

Do we have any other way to not auto-complete the text of the suggestion item when we click it?

Best regards,
Alex Yan

**Comment by hristop** [comment]
Internal BLI was created: BGSOFUIRILA-3698

**Comment by nikoletavnv** [comment]
Hi @yzy415 , 
We have created a way to suppress the autocomplete of input value when suggestion item is selected from the input suggestions list. You can do that by listening for the "suggestion-item-select" event and call preventDefault of that event. You have to disable the typeAhead of the input control first by setting no-typeahead="true". The change will be released with version 1.21 of the components.

Here is an example how you can achieve it for particular items: 

```
document.getElementById(<INPUT_CONTROL_ID>).addEventListener("ui5-suggestionItemSelect", (e) => {
const selectedItemText = e.detail.item.text || e.detail.item.textContent;
   if (selectedItemText === <ITEM_THAT_YOU_WILL_NOT_POPULATE>) {
        e.preventDefault();
       // Add your logic here for setting the value of the input control
      // Or leave it like that, to preserve the text you have entered by typing in the input
   }
});
```
Best regards, 
Nikoleta
